node/p2p/addrs.go

```diff
@@ -0,0 +1,59 @@
+package p2p
+
+import (
+	"fmt"
+
+	"github.com/libp2p/go-libp2p-core/host"
+	p2pconfig "github.com/libp2p/go-libp2p/config"
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+func Listen(listen []string) interface{} {
+	return func(host host.Host) (err error) {
+		maListen := make([]ma.Multiaddr, len(listen))
+		for i, addr := range listen {
+			maListen[i], err = ma.NewMultiaddr(addr)
+			if err != nil {
+				return fmt.Errorf("failure to parse config.P2P.ListenAddresses: %s", err)
+			}
+		}
+		return host.Network().Listen(maListen...)
+	}
+}
+
+func AddrsFactory(announce []string, noAnnounce []string) interface{} {
```

Comment:
Same question here, why return an interface ? Is this something required by libp2p ?